File: 413-Arithmetic-Slices/numberOfArithmeticSlices.java
class Solution {
  public int numberOfArithmeticSlices(int[] A) {
    int num_slices = 0;
    int add = 0;
    
    for (int i = 2; i < A.length; i++) {
      if (A[i] - A[i-1] == A[i-1] - A[i-2]) {
        add++;
        num_slices += add;
      }
      else {
        add = 0;
      }
    }
    
    return num_slices;
  }
}

File: 413-Arithmetic-Slices/numberOfArithmeticSlices.py
class Solution:
  def numberOfArithmeticSlices(self, A: List[int]) -> int: 
    num_slices = 0
    add = 0
    
    for i in range(2, len(A)):
      if (A[i] - A[i-1]) == (A[i-1] - A[i-2]):
        add += 1
        num_slices += add
      else:
        add = 0
    return num_slices